source/documentation/runbooks/004-analytical-platform-compute-maintenance.html.md.erb

```diff
@@ -0,0 +1,116 @@
+- Use account `modernisation-platform-developer` for Test and Procuction and `modernisation-platform-sandbox` for Development.
+- If necessary update `~/kube/config` as follows `aws eks update-kubeconfig --region eu-west-2 --name analytical-platform-compute-test`.
+- Set context as follows  `kubectl config use-context arn:aws:eks:eu-west-2:767397661611:cluster/analytical-platform-compute-test`.
+
+
+**Note:** amend above appropriately for the environment you are working in.
+
+## Impact on Users
+
+As this is a live service there could be an impact on users so this will have to be taken into consideration when planning the work.
+
+The impact on users depends on what is planned to be upgraded/patched.
+
+For example
+If upgrading `cloudwatch logs` agent, the user impact is minimal, applications will run, logs might be delayed and you will not require a maintenance window.
+
+IF upgrading `karpenter`, the user impact is high because jobs might not schedule as expected so you will have to agree when to schedule a maintenance window.
+
+To schedule a maintenance window go to [Pagerduty Maintenance Page](https://moj-digital-tools.pagerduty.com/external-status-dashboard/P1E3FLK/maintenance?tab=ongoing) and use the `Post Maintenance` button.
+
+If in doubt discuss with the team.
+
+
+## Example Pull requests
+
+[Upgrade Pull Request](https://github.com/ministryofjustice/analytical-platform/pull/5795)
+
+[Patching Pull Request](https://github.com/ministryofjustice/modernisation-platform-environments/pull/7548)
+
+##  Upgrade the EKS Control Plane
+
+1. Update the `eks_cluster_version` to the new version in `terraform/environments/analytical-platform-compute/environment-configuration.tf`.
+1. Commit and push your results to the branch.
+1. Check the Terraform plan.
+1. If plan as expected request approval of the PR.
+
+
+
+##  Upgrade the EKS Nodes
+
+1. Check the eks_node_version from environment-configuration.tf against the bottlerocket [changelog](https://github.com/bottlerocket-os/bottlerocket/blob/develop/CHANGELOG.md) to see if a new version is available.
+1. If so the `eks_node_version` is formed from '${BOTTLEROCKET_OS_RELEASE}]-${FIRST_EIGHT_CHARACTERS_OF_RELEASE_SHA}' i.e  [12.5.0](https://github.com/bottlerocket-os/bottlerocket/releases/tag/v1.25.0)-[388e1050](https://github.com/bottlerocket-os/bottlerocket/commit/388e1050a669dd2544007f2af336832b68fa0d64). Tip look at the url.
+1. Update `eks_node_version` in `environment-configuration.tf` with the value from above.
+1. Commit and push your results to the branch.
+1. Check the Terraform plan.
+1. If plan as expected request approval of the PR.
+
+##  Upgrade the EKS add-ons
+
+1. Run the following command and interpret the results to understand what version the add-ons should be upgraded to: `aws eks describe-addon-versions` Output to a text file (> file.txt) if you want to interpret the content more easily.
```

Comment:
Can we link to the relevant AWS docs here somewhere in case this changes in future?

Reply:
``` suggestion 
1. Run the following command and interpret the results to understand what version the add-ons should be upgraded to: `aws eks describe-addon-versions`. Output to a text file (`> file.txt`) if you want to interpret the content more easily.